### ISSUE TITLE
Wrong package version displayed

### ISSUE THREAD

**Issue by 130s** [ros-infrastructure/rosindex#301]
About `moveit_setup_assistant` along with `moveit2` on `Galatic`:
- [rosdistro shows 2.3.4](https://github.com/ros/rosdistro/blob/ae8032caf28c3340e2f8120b54bccab66d28e293/galactic/distribution.yaml#L2362)
- [index shows 2.7.3](https://index.ros.org/p/moveit_setup_assistant/github-ros-planning-moveit2/#galactic) (which seems to be the [largest version (moveit2/tags)](https://github.com/ros-planning/moveit2/tags) across all released distros as of today).
   ![2023-04-25_moveit2_index ros org](https://user-images.githubusercontent.com/1840401/234397312-aa60b33b-9e86-4e94-b313-4df40b392482.png)

**Comment by tfoote** [comment]
I think that the package version is being picked out of the source entry and it appears that moveit_setup_assistant has `main` registered as the development branch for all distros even back to galactic. It would probably be good to show that specifically separately than the rosdistro version to provide visibility to unreleased versions.

In the short term though I'd recommend considering suggesting that moveit_setup_assistant consider having development branches for the releases.

**Comment by 130s** [comment]
> It would probably be good to show that specifically separately than the rosdistro version to provide visibility to unreleased versions.

That certainly helps some usecases, maybe among advanced developers.

However, I'd imagine majority of usecases for version showing feature are for the packages installed via pre-built installer (deb for Ubuntu), where there should be only 1 version for each distro and that is the version released into each distro. So I'd say this version feature should show the released version.

**Comment by tfoote** [comment]
I agree, that's why I didn't close this. A pull request would be appreciated to make this happen but as it's a corner case right now for devel branches that are out of sync with the releases it's not something that we can prioritize.
